DigitalLearningSolutions.Web/Scripts/trackingSystem/reports.ts

```diff
@@ -33,8 +38,25 @@ function generateChart(request: XMLHttpRequest) {
   if (typeof request.response === 'string') {
     response = JSON.parse(response);
   }
-  const data = constructChartistData(response);
+  chartData = constructChartistData(response);
+  drawChartOrDataPointMessage();

```

Comment:
I think it makes more sense split the functions as 
* Save chart data
* Draw the graph

It's a natural split, but we're doing one once and the other repeatedly.
So I would move this function call out into `request.onload` and rename `generateChart` to `saveChartData`. I'd probably change the param to just take in the response instead of the whole request.

Reply:
I'm not sure I can change the parameter due to the strange typing that is browser dependent (IE). This method is what handles that.

Reply:
Hmm, that is very strange. Fair, let's leave it then.